### PR TITLE
[ELYWEB-1] Port the Undertow / Servlet Integration to Elytron Web

### PR DESCRIPTION
**Pull request by darranl**
As WildFly Elytron was integrated with Undertow we started with an integration project integrating WildFly Elytron with the core of Undertow, this is as Elytron Web exists before this PR.

As integration with the application server and servlet container progressed more specialised integration was required within the Undertow subsystem, this started off as an occasionally tweak until the changes were quite large.

This pull request pulls those changes out of the Undertow subsystem into a separate module to hold the Undertow / Servlet integration.

### CODE REVIEW

File: undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ElytronHttpServletExchange.java [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        final ServletRequestContext servletRequestContext = httpServerExchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY);
+        ServletRequest req = servletRequestContext.getServletRequest();
+        ServletResponse resp = servletRequestContext.getServletResponse();
+        RequestDispatcher disp = req.getRequestDispatcher(path);
+
+        final FormResponseWrapper respWrapper = httpServerExchange.getStatusCode() != OK && resp instanceof HttpServletResponse
+                ? new FormResponseWrapper((HttpServletResponse) resp) : null;
+
+        try {
+            disp.forward(req, respWrapper != null ? respWrapper : resp);
```
Comment (mchoma) [inline undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ElytronHttpServletExchange.java]: Coverity is complaining 

https://u2389337.ct.sendgrid.net/wf/click?upn=08onrYu34A-2BWcWUl-2F-2BfV0V05UPxvVjWch-2Bd2MGckcRYZquIFszCcwQkbx6rX2y-2BUxNCGJdkIRvpY0lX6FsmGlUNLm1ZLqjmn3Xwt68CpLFqJgqd9fwu9t55-2Fv2R1w01q_2fPkMU1miRf-2FtjnMal9VVM35TqEfZozw4WTtk2YIMAKSiRrskMGOQZz0Ta8rCKNvNs3WUbHVZvDmaDHeD309T5r6-2BsMVqQB-2BO6BuefCCb4s1a50ellRFmuTIofeoY143RBQ8RnqVwL-2F1meA8j0sJepALas4QiG2qruipA-2B4k4D6BX9I62R4IWYXHSEAOTESI8QP9dueBxyBpGXjdQ3yhqw-3D-3D

Comment (mchoma) [reply undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ElytronHttpServletExchange.java]: https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getRequestDispatcher(java.lang.String)

Comment (darranl) [reply undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ElytronHttpServletExchange.java]: Do you have details of the error?  To click that link I need to grant loads of permissions just to see it.

Comment (mchoma) [reply undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ElytronHttpServletExchange.java]: Javadoc for ServletRequest.getRequestDispatcher(path) says null can be returned. Therefore Coverity detect disp.forward() as potentially dereferencing null code.

Comment (darranl) [reply undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ElytronHttpServletExchange.java]: I think if we can get an issue created here we can take a look https://issues.jboss.org/projects/ELYWEB/summary the real objective of this step was to pull in the code as-is as much as possible to form the baseline of the porting - we currently don't have any specific error messages for elytron web yet but we will likely need them for this.

Comment (mchoma) [reply undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ElytronHttpServletExchange.java]: I have created https://issues.jboss.org/browse/ELYWEB-6

**Pull request merged**